Eco.EM.Framework/GroupsSystem/GroupsData.cs

```diff
@@ -53,10 +53,11 @@ public bool DeleteGroup(string dirtyGroupName)
             return Groups.Remove(group);
         }
 
-        public SimpleGroupUser GetGroupUser(User user)
-        {
-            return AllUsers.FirstOrDefault(entry => entry.Name == user.Name || entry.SlgID == user.SlgId || entry.SteamID == user.SteamId);
```

Comment:
Since username is always the same and never null we can just do:

entry.Name == user.Name && (entry.SlgID == user.SlgId || entry.SteamID == user.SteamId)

as we only need to check for the slgid being null or empty and the steam id

Reply:
These have been implemented as requested in all instances changed by me.

Reply:
Kye - in regards to Discord comments on same names and related, here is the part of source code I mention:
```cs
 public User PrepareNewUser(string steamId, string slgId, string username)
        {
            // Create new User
            foreach (char c in usernameCharBlacklist)
                username = username.Replace(c, '?');

            // Don't let them have only numbers since they can spoof an SLG ID that way.
            if (username.Length > 10 && username.IsDigitsOnly())
                username = $"Eco{username}";

            // Don't let them use slg so they cant spoof an SLG ID
            if (username.StartsWith("slg"))
                username = $"Eco{username}";

            // Use predefined "Unknown" name for some cases
            var noSpaceName = username.RemoveSpaces();
            if (!username.IsSet() || !noSpaceName.IsSet() || // empty
                (noSpaceName.Length <= 2 && noSpaceName.All(char.IsPunctuation))) // short punctuation only
                username = unnamedUsername;

            var i = 2;
            // Clean profanity before saving name
            username = username.ProfanityFiltered();
            while (UsersByDisplayName.Any(x => x.Key.CompareCaseInsensitive(username) == 0))
            {
                // There is an existing user with the same name.  Append a suffix to this
                // username to make it unique (applies to this server only)
                username = $"{username}#{i}";
                i++;
            }

            var storageHandle = StorageManager.Obj.CreateStorageHandle("Users", $"{steamId}_{slgId}");
            var user          = User.CreateUser(username, steamId, slgId, storageHandle);
            var registrar     = Registrars.Get<User>();
            AddUser(user);

            // setting up inventories
            this.ResetUser(user);
            user.Position = GetOrRelocateSpawnPosition();
            registrar.Insert(user);
            this.AddTickable(user); // no ticks for internal users
            user.MarkDirty();

            this.aliases.Add(user);
            NewUserJoinedEvent.Invoke(user);

            return user;
        }
```